From 22e0523e4f86081498f5154c018177ccbdfa6ca1 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Thu, 10 Jul 2014 01:02:12 -0400
Subject: [acpi] hotplug/pci: Drop handle field from struct acpiphp_bridge

Message-id: <20140710010212.22146.91878.stgit@gir.stowe>
Patchwork-id: 85577
O-Subject: [RHEL7.1 PATCH 038/105] ACPI / hotplug / PCI: Drop handle field from struct acpiphp_bridge
Bugzilla: 1114228
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1114228
Upstream Status: 89373a55d294b53e85792dbc636015b83d492f67

commit 89373a55d294b53e85792dbc636015b83d492f67
Author: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Date:   Sat Jul 13 23:27:25 2013 +0200

    ACPI / hotplug / PCI: Drop handle field from struct acpiphp_bridge

    The handle field in struct acpiphp_bridge is only used by
    acpiphp_enumerate_slots(), but in that function the local handle
    variable can be used instead, so make that happen and drop handle
    from struct acpiphp_bridge.

    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
    Tested-by: Mika Westerberg <mika.westerberg@linux.intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/hotplug/acpiphp.h b/drivers/pci/hotplug/acpiphp.h
index 699732b..6099d79 100644
--- a/drivers/pci/hotplug/acpiphp.h
+++ b/drivers/pci/hotplug/acpiphp.h
@@ -67,7 +67,6 @@ struct acpiphp_bridge {
  struct list_head list;
  struct list_head slots;
  struct kref ref;
- acpi_handle handle;
 
  struct acpiphp_context *context;
 
diff --git a/drivers/pci/hotplug/acpiphp_glue.c b/drivers/pci/hotplug/acpiphp_glue.c
index fb33ac5..60b5c54 100644
--- a/drivers/pci/hotplug/acpiphp_glue.c
+++ b/drivers/pci/hotplug/acpiphp_glue.c
@@ -1084,7 +1084,6 @@ void acpiphp_enumerate_slots(struct pci_bus *bus)
 
  INIT_LIST_HEAD(&bridge->slots);
  kref_init(&bridge->ref);
- bridge->handle = handle;
  bridge->pci_dev = pci_dev_get(bus->self);
  bridge->pci_bus = bus;
 
@@ -1124,10 +1123,10 @@ void acpiphp_enumerate_slots(struct pci_bus *bus)
  mutex_unlock(&bridge_mutex);
 
  /* register all slot objects under this bridge */
- status = acpi_walk_namespace(ACPI_TYPE_DEVICE, bridge->handle, 1,
+ status = acpi_walk_namespace(ACPI_TYPE_DEVICE, handle, 1,
          register_slot, NULL, bridge, NULL);
  if (ACPI_FAILURE(status)) {
-  acpi_handle_err(bridge->handle, "failed to register slots\n");
+  acpi_handle_err(handle, "failed to register slots\n");
   cleanup_bridge(bridge);
   put_bridge(bridge);
  }
-- 
1.7.1